backend/app/modules/common/sablier_events_mapper.py

```diff
@@ -0,0 +1,169 @@
+    copy_event_items = deepcopy(event_items)
+
+    if inclusively is True:
+        to_timestamp += 1
+
+    for item in event_items:
+        if from_timestamp and item["timestamp"] < from_timestamp:
+            copy_event_items.remove(item)
+        if to_timestamp and item["timestamp"] > to_timestamp:
+            copy_event_items.remove(item)
+
+    return copy_event_items
+
+
+def _process_create(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    amount = int(action["amountA"]) if action["amountA"] else 0
+    deposit_before = starting_deposit
+    starting_deposit += amount
+    lock_item = SablierEventLock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.LOCK.value,
+        amount=amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.CREATE,
+    )
+    return lock_item, starting_deposit
+
+
+def _process_withdraw(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    amount = int(action["amountB"]) if action["amountB"] else 0
+    deposit_before = starting_deposit
+    starting_deposit -= amount
+    lock_item = SablierEventUnlock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.UNLOCK.value,
+        amount=amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.WITHDRAW,
+    )
+    return lock_item, starting_deposit
+
+
+def _process_cancel(
+    action: SablierAction, starting_deposit: int
+) -> Tuple[SablierEvent, int]:
+    intact_amount = int(action["amountB"]) if action["amountB"] else 0
+    cancelled_amount = int(action["amountA"]) - intact_amount
+    deposit_before = starting_deposit
+    starting_deposit = intact_amount
+    lock_item = SablierEventUnlock(
+        __source=DepositSource.SABLIER,
+        __typename=EventType.UNLOCK.value,
+        amount=cancelled_amount,
+        timestamp=int(action["timestamp"]),
+        transactionHash=action["hash"],
+        depositBefore=deposit_before,
+        user=action["addressB"],
+        type=SablierEventType.CANCEL,
+    )
+    return lock_item, starting_deposit
+
+
+def _convert(actions: List[SablierAction]) -> List[SablierEvent]:
+    lock_items = []
+    action_strategy = {
+        SablierEventType.CREATE.value: _process_create,
+        SablierEventType.WITHDRAW.value: _process_withdraw,
+        SablierEventType.CANCEL.value: _process_cancel,
+    }
+    starting_deposit = 0
+
+    for action in actions:
+        category = action["category"]
+        if category in action_strategy:
+            process_func = action_strategy[category]
+            lock_item, starting_deposit = process_func(action, starting_deposit)
+            lock_items.append(lock_item)
```

Comment:
It's still a static one but placed in the hardcoded dict. I'm not a big fan of doing if-chaining in the code and did it on purpose to keep the code more Open/Closed here. More Open/Closed because if we wanted to add dealing with another option, then we wouldn't be required to change the whole ifology but only add a new function with key to the dictionary. But if you stick to your caveat, then I can change it :v: